external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -1702,28 +1702,26 @@ impl PathExpander for Move2024PathExpander {
         sp!(loc, avalue_): P::AttributeValue,
     ) -> Option<E::AttributeValue> {
         use E::AttributeValue_ as EV;
-        use P::{AttributeValue_ as PV, LeadingNameAccess_ as LN, NameAccessChain_ as PN};
+        use P::AttributeValue_ as PV;
         Some(sp(
             loc,
             match avalue_ {
                 PV::Value(v) => EV::Value(value(context, v)?),
-                PV::ModuleAccess(
-                    sp!(ident_loc, PN::Two(sp!(aloc, LN::AnonymousAddress(a)), n)),
-                ) => {
-                    let addr = Address::anonymous(aloc, a);
-                    let mident = sp(ident_loc, ModuleIdent_::new(addr, ModuleName(n)));
-                    if context.module_members.get(&mident).is_none() {
-                        context.env.add_diag(diag!(
-                            NameResolution::UnboundModule,
-                            (ident_loc, format!("Unbound module '{}'", mident))
-                        ));
-                    }
-                    EV::Module(mident)
-                }
-                // TODO consider if we want to just force all of these checks into the well-known
-                // attribute setup
+                // A bit strange, but we first try to resolve it as a term, if that fails, we try
+                // to resolve it as a module access.
                 PV::ModuleAccess(access_chain) => {
-                    match self.resolve_name_access_chain(context, Access::Term, access_chain) {
+                    let result =
+                        self.resolve_name_access_chain(context, Access::Term, access_chain.clone());
```

Comment:
IMO this should try to resolve Term and Module and accept either if there is one, and error if both resolve.

Reply:
Done, but let me know if you can come up with some other tests where we hit an ambiguity